src/Microsoft.DocAsCode.MarkdownLite/JsonRenderer.cs

```diff
@@ -0,0 +1,305 @@
+        {
+            return this.Insert("Br", this.GetSize(), Type.Size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownTextToken token, MarkdownInlineContext context)
+        {
+            return this.Insert("InLineText(" + token.Content.Replace("\n", "\\n") + ")", this.GetSize(), Type.Size);
+        }
+
+        #endregion
+
+        #region Misc
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownIgnoreToken token, IMarkdownContext context)
+        {
+            return this.Insert("Ignore", this.GetSize(), Type.Size);
+        }
+
+        public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownRawToken token, IMarkdownContext context)
+        {
+            switch (token.Rule.Name)
+            {
+                case ("Html"):
+                {
+                    return this.Insert("Raw(FromHtml)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.AutoLink"):
+                {
+                    return this.Insert("Raw(FromInLineAutoLink)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.CodeElement"):
+                {
+                    return this.Insert("Raw(FromInLineCodeElement)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.Comment"):
+                {
+                    return this.Insert("Raw(FromInLineComment)", this.GetSize(), Type.Size);
+                }
+                case ("GfmHtmlComment"):
+                {
+                    return this.Insert("Raw(FromGfmHtmlComment)", this.GetSize(), Type.Size);
+                }
+                case ("Inline.Gfm.Url"):
+                {
+                    return this.Insert("Raw(FromGfmUrl)", this.GetSize(), Type.Size);
+                }
+                default:
+                    return this.Insert("Raw", this.GetSize(), Type.Size);
+            }
+        }
+
+        #endregion
+
+        #region Private
+
+        private StringBuffer Insert(string name, string tokenContent, Type tokenType)
+        {
+            var result = (StringBuffer) $"{{\"name\":\"{name}\"";
+            switch (tokenType)
+            {
+                case Type.Child:
+                {
+                    // If tokenContent is not empty ,should remove the last character(',')
+                    if (tokenContent.Length > 0)
+                    {
+                        tokenContent = tokenContent.Remove(tokenContent.Length - 1);
+                    }
+                    result += $",\"children\":[{tokenContent}]}},";
+                    break;
+                }
+                case Type.Size:
+                {
+                    // size will be needed later
+                    // result += "\"size\":" + tokenContent + "\n},";
+                    result += "},";
```

Comment:
Size is a parameter of node to Show the tokentree ,it will needed and defined later but I can remove it now.

---
In reply to: [74728604](https://github.com/dotnet/docfx/pull/608#discussion_r74728604) [](ancestors = 74728604)